Assets/Scripts/Player/PlayerController.cs

```diff
@@ -111,7 +120,54 @@ private void Update()
             Vector3 playerPosition = transform.position;
             _camera.transform.position = playerPosition + _cameraPosition;
             _camera.transform.LookAt(playerPosition);
+
+            // Check and makes character facing the target object when moving toward it
+            if (target != null) FaceTarget(target.gameObject, true, out _);
```

Comment:
why do you set updateRotation to false below? This line is just needed because of that.

Reply:
with implementation of an InteractLogic class not needed anymore

Reply:
Yeah just removed updateRotation. Seems like it is not needed